packages/api/src/router/kubernetes/router/namespaces.ts

```diff
@@ -0,0 +1,40 @@
+import { TRPCError } from "@trpc/server";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+
+import type { KubernetesNamespace, KubernetesNamespaceState } from "@homarr/definitions";
+import { logger } from "@homarr/log";
+
+import { kubernetesMiddleware } from "../../../middlewares/kubernetes";
+import { createTRPCRouter, permissionRequiredProcedure } from "../../../trpc";
+import { KubernetesClient } from "../kubernetes-client";
+
+dayjs.extend(relativeTime);
+
+export const namespacesRouter = createTRPCRouter({
+  getNamespaces: permissionRequiredProcedure
+    .requiresPermission("admin")
+    .unstable_concat(kubernetesMiddleware())
+    .query(async (): Promise<KubernetesNamespace[]> => {
+      const { coreApi } = KubernetesClient.getInstance();
+
+      try {
+        const namespaces = await coreApi.listNamespace();
+
+        return namespaces.items.map((namespace) => {
+          return {
+            status: namespace.status?.phase as KubernetesNamespaceState,
+            name: namespace.metadata?.name ?? "unknown",
+            creationTimestamp: dayjs().to(dayjs(namespace.metadata?.creationTimestamp)),
```

Comment:
Not sure if it makes sense to do this here as we don't know the locale of the user and so it will not be translated

Reply:
make sense to me.